presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1758,21 +1761,30 @@ private static Number toNumber(Object value)
     {
         private final StatementClient client;
         private final Consumer<QueryStats> progressCallback;
+        private final WarningsManager warningsManager;
+        private final String updateType;
 
-        private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progressCallback)
+        private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progressCallback, WarningsManager warningsManager)
         {
             this.client = requireNonNull(client, "client is null");
             this.progressCallback = requireNonNull(progressCallback, "progressCallback is null");
+            this.warningsManager = requireNonNull(warningsManager, "warningsManager is null");
+            this.updateType = client.isRunning() ? client.currentStatusInfo().getUpdateType() : client.finalStatusInfo().getUpdateType();
         }
 
         @Override
         protected Iterable<List<Object>> computeNext()
         {
+            if (updateType == null) {
```

Comment:
Maybe keep track of whether this is a query more explicitly. Instead of checking for `updateType == null`, record a "isQuery" flag in the constructor and test for that.